src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -956,7 +956,7 @@ private static void RemoveDuplicateInferredTupleNames(ArrayBuilder<string> infer
 
         private static string InferTupleElementName(ExpressionSyntax element)
         {
-            SyntaxToken nameToken = element.ExtractAnonymousTypeMemberName();
+            SyntaxToken nameToken = element.TryGetInferredMemberName();
             if (nameToken.Kind() == SyntaxKind.IdentifierToken)
```

Comment:
There is at least one case where we needed the token. I made the public API to return a string, and kept the internal API returning a token.